specs/networking/p2p-interface.md

```diff
@@ -0,0 +1,760 @@
+  - [Transport](#transport-1)
+  - [Multiplexing](#multiplexing-1)
+  - [Protocol Negotiation](#protocol-negotiation-1)
+  - [Encryption](#encryption)
+  - [Gossipsub](#gossipsub)
+  - [Req/Resp](#reqresp)
+  - [Discovery](#discovery)
+  - [Compression/Encoding](#compressionencoding)
+- [libp2p Implementations Matrix](#libp2p-implementations-matrix)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+# Network Fundamentals
+
+This section outlines the specification for the networking stack in Ethereum 2.0 clients.
+
+Sections that have differing parameters for mainnet launch and interoperability testing are split into subsections. Sections that are not split have the same parameters for interoperability testing as mainnet launch.
+
+## Transport
+
+Even though libp2p is a multi-transport stack (designed to listen on multiple simultaneous transports and endpoints transparently), we hereby define a profile for basic interoperability.
+
+#### Interop
+
+All implementations MUST support the TCP libp2p transport, and it MUST be enabled for both dialing and listening (i.e. outbound and inbound connections).
+
+The libp2p TCP transport supports listening on IPv4 and IPv6 addresses (and on multiple simultaneously). Clients SHOULD allow the operator to configure the listen IP addresses and ports, including the addressing schemes (IPv4, IPv6).
+
+To facilitate connectivity, and avert possible IPv6 routability/support issues, clients participating in the interoperability testnet MUST expose at least ONE IPv4 endpoint.
+
+All listening endpoints must be publicly dialable, and thus not rely on libp2p circuit relay, AutoNAT or AutoRelay facilities.
+
+Nodes operating behind a NAT, or otherwise undialable by default (e.g. container runtime, firewall, etc.), MUST have their infrastructure configured to enable inbound traffic on the announced public listening endpoint.
+
+#### Mainnet
+
+All requirements from the interoperability testnet apply, except for the IPv4 addressing scheme requirement.
+
+At this stage, clients are licensed to drop IPv4 support if they wish to do so, cognizant of the potential disadvantages in terms of Internet-wide routability/support. Clients MAY choose to listen only on IPv6, but MUST retain capability to dial both IPv4 and IPv6 addresses.
+
+Usage of circuit relay, AutoNAT or AutoRelay will be specifically re-examined closer to the time.
+
+## Encryption and identification
+
+#### Interop
+
+[SecIO](https://github.com/libp2p/specs/tree/master/secio) with `secp256k1` identities will be used for initial interoperability testing.
+
+The following SecIO parameters MUST be supported by all stacks:
+
+-  Key agreement: ECDH-P256.
+-  Cipher: AES-128.
+-  Digest: SHA-256.
+
+#### Mainnet
+
+[Noise Framework](http://www.noiseprotocol.org/) handshakes will be used for mainnet. libp2p Noise support [is in the process of being standardised](https://github.com/libp2p/specs/issues/195) in the libp2p project.
```

Comment:
I think you should specify that mainnet clients MUST not connect via secio

Reply:
We ensure that peers at least support the Noise handshake - naturally clients will gravitate towards it as it will have the most peers available. That said, we want to allow alternate transports and protocols for environments where the default is not feasible.

The expectation is that libp2p clients will gradually be deprioritizing connections through secio naturally as better alternatives take its place (fewer round trips, better security) - there's no need to explicitly forbid it.